dask/array/reshape.py

```diff
@@ -226,6 +230,19 @@ def reshape(x, shape, merge_chunks=True):
         x = x.rechunk({i: 1 for i in range(din - dout)})
 
     inchunks, outchunks = reshape_rechunk(x.shape, shape, x.chunks)
+    # Check output chunks are not too large
+    max_chunksize_in_bytes = np.prod([max(i) for i in outchunks]) * x.dtype.itemsize
+    if limit is None:
```

Comment:
Just a heads up that the automatic splitting in https://github.com/dask/dask/pull/6514 broke some user's code and was partially reverted in https://github.com/dask/dask/pull/6665. That added a config option, `array.slicing.split-large-chunks`, to control whether the splitting happens automatically.

It might be worth skimming https://github.com/dask/dask/issues/6646 to see the type of behavior that was broken to judge if this PR would have similar issues (I haven't looked closely)

Reply:
I did not know about that, thanks for pointing that out!

Reply:
I've changed the condition to respect the `array.slicing.split-large-chunks` setting. I now have:

```python
    if (
        limit is None
        and config.config["array"]["slicing"]["split-large-chunks"] is not False
    ):
    # ... code continues ....
```